src/domain/app/layout/__tests__/PageLayout.test.tsx

```diff
@@ -13,54 +12,53 @@ import {
 } from '../../../../utils/testUtils';
 import PageLayout from '../PageLayout';
 
+jest.mock('../../../../generated/graphql-cms', () => ({
+  esModule: true,
+  ...jest.requireActual<Record<string, unknown>>(
+    '../../../../generated/graphql-cms'
+  ),
+  useNotificationQuery: jest.fn(),
+}));
+
 const notificationContent = 'Notification content';
 const notificationTitle = 'Notification title';
 
-const createMocks = (
-  notificationTitle: string,
-  notificationContent: string
-): MockedResponse[] => [
-  ...emptyMenuQueryMocks,
-  {
-    request: {
-      query: NotificationDocument,
-      variables: {
-        language: 'fi',
-      },
-    },
-    result: {
-      data: {
-        notification: fakeNotification({
-          title: notificationTitle,
-          content: notificationContent,
-        }),
-      },
-    },
-  },
-];
-
-const mocksWithNonEmptyNotification = createMocks(
-  notificationTitle,
-  `<p>${notificationContent}</p>`
-);
-const mocksWithEmptyNotification = createMocks('', '');
-
 it('PageLayout matches snapshot', () => {
+  const notificationMock = {
+    notification: fakeNotification({
+      title: '',
+      content: '',
+    }),
+  };
+  jest.spyOn(GraphQLCms, 'useNotificationQuery').mockReturnValue({
+    data: { ...notificationMock },
+    loading: false,
+  } as any);
   const { container } = render(
     <PageLayout>
       <div>Page layout children</div>
     </PageLayout>,
-    { mocks: mocksWithNonEmptyNotification }
+    { mocks: [...emptyMenuQueryMocks] }
   );
   expect(container.firstChild).toMatchSnapshot();
```

Comment:
Why did the snapshot here not change as before there was a non-empty notification here and now there is an empty notification here? PageLayout → Header → HeaderNotification → Notification so it should be there, or is e.g. PageLayout's render's firstChild somehow omitting it / stepping into some part of the layout that does not contain the notification?

Reply:
PageLayout.test.tsx.snap doesn't have any notification data in it so that's why it's matching. Maybe the non-empty notification mock didn't really work before?

Reply:
the mocks included menu and notification queries. Now the mocks includes only the menu query but the notification is mocked with spy. The result is exactly the same. The spy was needed, because now when I have given the client as an attribute to the query, it's not the same as the mocked apollo client in tests.

Reply:
We had a talk, and we came to a consensus that there was probably something wrong here. But it is ok to have the empty notification in this test case as there's a test case for the notification being there and not being there later.